skaled/main.cpp

```diff
@@ -1703,6 +1762,16 @@ int main( int argc, char** argv ) {
             }
         }
 
+#ifdef MIRAGE
+
+        // Configuring current group if it is regular syncing from catchup.
+        // Setting back correct group if it starting from snapshot mode.
+
+        uint64_t latestBlockTs = BlockChain::getLatestBlockTimestamp( *chainParams, getDataDir() );
```

Comment:
isn't it better to `updateIfNeed` under the `if (download snapshot)` ?
for regular start we update the group in the very begining, no need to update it second time